api/v1alpha1/properties.go

```diff
@@ -80,3 +80,15 @@ func (d *Device) IsUpdatedToFleetSpec(f *Fleet) bool {
 	}
 	return d.IsUpdatedToDeviceSpec() && deviceTemplateVersion == fleetTemplateVersion
 }
+
+// IsDecommissioning() is true if the device has added a DeviceDecommissioning ConditionType to its Conditions.
+func (d *Device) IsDecommissioning() bool {
+	if d.Status == nil || d.Status.Conditions == nil {
+		return false
+	}
+
+	if found := FindStatusCondition(d.Status.Conditions, DeviceDecommissioning); found == nil {
+		return false
+	}
+	return true
```

Comment:
do you need to validate the status or reason or is the existence enough?

https://github.com/flightctl/flightctl/blob/915f4c0596b7216683e309ee6b7c3f4d63a3b166/api/v1alpha1/properties.go#L42

Reply:
Currently if the Condition exists then that is the signal that the device has switched to Decommissioning, so we don't need other checks to determine it.

Reply:
OK I just figured that the condition existed but the value was `ConditionStatusFalse` by default.

Reply:
Yep it makes sense now! Updated